code-challenges.test.js

```diff
@@ -13,39 +13,101 @@
+        const object1 = { number: 15 }
+        const object2 = { number: 0 }
+        const object3 = { number: -7 }
+        // Expected output: "15 is divisible by three"
+        // Expected output: "0 is divisible by three"
+        // Expected output: "-7 is not divisible by three"
+        expect(divisibleByThree(object1)).toEqual("15 is divisible by three")
+        expect(divisibleByThree(object2)).toEqual("0 is divisible by three")
+        expect(divisibleByThree(object3)).toEqual("-7 is not divisible by three")
+    })
+})
+
+// ReferenceError: divisibleByThree is not defined
+
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// // Pseudo code:
+const divisibleByThree = (object) => {
+    // iterate with if statement using modulo 3
+    // compare remainder of each object to 0
+    if (object.number % 3 === 0) {
+        return `${object.number} is divisible by three`
+        // return if object number is divisible by three or not
+      } else {
+        return `${object.number} is not divisible by three`
+      }
+}
+  
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+describe("wordsCapitalized", () => {
+    it("takes in an array of words and returns an array with all the words capitalized", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+        expect(wordsCapitalized(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(wordsCapitalized(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+
+// ReferenceError: wordsCapitalized is not defined
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
\ No newline at end of file
+// Pseudo code:
+// input: array of random nouns
+// output: array of random nouns with capitalizations
+const wordsCapitalized = (array) => {
+    // iterate with .map
+    return array.map((string) => {
+        // using .charAt and .toUpperCase identify and capitalize the first letter in each string
+        // using .slice join the rest of the word in each string to the array
+       return string.charAt(0).toUpperCase() + string.slice(1)
```

Comment:
The wording "join the rest of the word in each string to the array" is inaccurate. You are concatenating the rest of the word to the capitalized first letter of the respective word, which then gets returned in the new array created by map.